src/main/java/org/killbill/billing/payment/api/Payment.java

```diff
@@ -84,4 +84,9 @@
      * @return the list of attempts on that payment
      */
     List<PaymentTransaction> getTransactions();
+
+    /**
+     * @return the next Scheduled payment attempt
+     */
+    PaymentAttempt getNextScheduledPaymentAttempt();
```

Comment:
Yes, it should be `List<PaymentAttempt> getPaymentAttempts();`

As described in http://docs.killbill.io/0.16/userguide_payment.html#_payment_control_plugins, payment attempts are not just linked to retries, but to control plugins in general. Also, they map to a transaction, not a payment, so a payment with two captures could have two trains of retries in parallel.

Note that the method would return the past (present in the `payment_attempts` table) and future attempts (future attempts would be the retries, which can be found in the `payment-service` notification queue).